packages/common/src/extensions/slickRowBasedEdit.ts

```diff
@@ -241,16 +246,26 @@ export class SlickRowBasedEdit {
     });
   }
 
-  translate() {
-    this.btnUpdateTitle = this.getTitleOrDefault('updateButtonTitle', 'Update the row');
-    this.btnEditTitle = this.getTitleOrDefault('editButtonTitle', 'Edit the Row');
-    this.btnDeleteTitle = this.getTitleOrDefault('deleteButtonTitle', 'Delete the Row');
-    this.btnCancelTitle = this.getTitleOrDefault('cancelButtonTitle', 'Cancel changes of the Row');
-
-    const viewport = this._grid.getViewport();
+  /**
+   * Translate with current locale when `enableTranslate` is set or use title texts provided by user.
+   * We want to translate only once to avoid having to translate on each dataset item, we'll just reuse the translations if it was already translated earlier.
+   * User could optionally force a retranslate even if it was already translated
+   * @param {Boolean} [forceRetranslate] - even if it was already translate, force a retranslate
+   */
+  translate(forceRetranslate = false) {
+    this._currentLang = this.extensionUtility.translaterService?.getCurrentLanguage() ?? 'en';
```

Comment:
if its just for readability we could turn it into a getter but I'd ultimately always fetch it from extensionUtility

Reply:
the reason I've done this way is first because that is exactly how you called it in the `getTitleOrDefault()`, if I refactor this one then it would be better to refactor both